addons/survey/tests/test_survey.py

```diff
@@ -159,3 +159,150 @@ def test_skipped_values(self):
+            'value': 'a',
+        })
+        page_with_description = self.env['survey.question'].create({
+            'title': 'with desc',
+            'survey_id': my_survey.id,
+            'sequence': 5,
+            'is_page': True,
+            'question_type': False,
+            'description': 'This page has a description',
+        })
+        multiple_choice_2 = self.env['survey.question'].create({
+            'title': 'multiple_choice with no answers',
+            'survey_id': my_survey.id,
+            'sequence': 7,
+            'is_page': False,
+            'question_type': 'multiple_choice',
+        })
+        text_box_2 = self.env['survey.question'].create({
+            'title': 'text_box with valid trigger',
+            'survey_id': my_survey.id,
+            'sequence': 8,
+            'is_page': False,
+            'question_type': 'text_box',
+            'triggering_question_id': multiple_choice_1.id,
+            'is_conditional': True,
+        })
+        matrix = self.env['survey.question'].create({
+            'title': 'matrix with invalid trigger (page)',
+            'survey_id': my_survey.id,
+            'sequence': 9,
+            'is_page': False,
+            'question_type': 'matrix',
+            'triggering_question_id': page_with_description.id,
+            'is_conditional': True,
+        })
+        simple_choice_2 = self.env['survey.question'].create({
+            'title': 'simple choice w/ invalid trigger (no suggested_answer_ids)',
+            'survey_id': my_survey.id,
+            'sequence': 10,
+            'is_page': False,
+            'question_type': 'simple_choice',
+            'triggering_question_id': multiple_choice_2.id,
+            'is_conditional': True,
+        })
+        simple_choice_3 = self.env['survey.question'].create({
+            'title': 'text_box w/ invalid trigger (not a mcq)',
+            'survey_id': my_survey.id,
+            'sequence': 11,
+            'is_page': False,
+            'question_type': 'simple_choice',
+            'triggering_question_id': text_box_2.id,
+            'is_conditional': True,
+            'suggested_answer_ids': False,
+        })
+        text_box_3 = self.env['survey.question'].create({
+            'title': 'text_box w/ invalid trigger (suggested_answer_ids is False)',
+            'survey_id': my_survey.id,
+            'sequence': 12,
+            'is_page': False,
+            'question_type': 'text_box',
+            'triggering_question_id': simple_choice_3.id,
+            'is_conditional': True,
+        })
+
+        invalid_records = page_without_description + text_box_1 + numerical_box \
+            + matrix + simple_choice_2 + simple_choice_3 + text_box_3
+        question_and_page_ids = my_survey.question_and_page_ids
+        returned_questions_and_pages = my_survey._get_pages_and_questions_to_show()
+
+        self.assertEqual(question_and_page_ids - invalid_records, returned_questions_and_pages)
```

Comment:
You were right by the way, it's more efficient to test it this way.
It's of course easier now that there is only 1 call ^^